src/test/scala/ch/epfl/bluebrain/nexus/tests/NewBaseSpec.scala

```diff
@@ -0,0 +1,194 @@
+            (_, response) => response.status shouldEqual StatusCodes.OK
+          }
+        }
+        succeed
+    }
+
+  def ensureRealmExists: Assertion = {
+    val body =
+      jsonContentOf(
+        "/iam/realms/create.json",
+        Map(
+          quote("{realm}") -> config.realm
+        )
+      )
+
+    cl.get[Json](s"/realms/$realmLabel", Identity.Anonymous) {
+      (json, response) =>
+        val rev = json.hcursor.downField("_rev")
+          .as[Int]
+          .toOption
+        Some(response.status) should contain oneOf (StatusCodes.OK, StatusCodes.NotFound)
+        rev match {
+          case Some(r) =>
+            cl.put[Json](s"/realms/$realmLabel?rev=$r", body, Identity.Anonymous) {
+              (_, response) =>
+                response.status shouldEqual StatusCodes.OK
+            }
+          case None =>
+            cl.put[Json](s"/realms/$realmLabel", body, Identity.Anonymous) {
+              (_, response) =>
+                response.status shouldEqual StatusCodes.Created
+            }
+        }
+    }
+  }
+
+  private[tests] implicit class HttpClientOps(val httpClient: UntypedHttpClient[Task]) {
+
+    def post[A](url: String, body: Json, identity: Identity)
+               (assertResponse: (A, HttpResponse) => Assertion)
+               (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(POST, url, Some(body), identity)(assertResponse)
+
+    def put[A](url: String, body: Json, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(PUT, url, Some(body), identity)(assertResponse)
+
+    def patch[A](url: String, body: Json, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(PATCH, url, Some(body), identity)(assertResponse)
+
+    def get[A](url: String, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(GET, url, None, identity)(assertResponse)
+
+    def delete[A](url: String, identity: Identity)
+                 (assertResponse: (A, HttpResponse) => Assertion)
+                 (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(DELETE, url, None, identity)(assertResponse)
+
+    def request[A](method: HttpMethod,
+                   url: String,
+                   body: Option[Json],
+                   identity: Identity)
+                  (assertResponse: (A, HttpResponse) => Assertion)
+                  (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      httpClient(
+        HttpRequest(
+          method = method,
+          uri = s"$baseUrl$url",
+          headers = userTokens.get(identity).fold(List.empty[Authorization])(List(_)),
+          entity = body.fold(HttpEntity.Empty)
```

Comment:
Few tests require a different Accept header (fetch a file metadata vs. fetch the file itself). But I guess you dealt with those in a different way.